source/agora/consensus/validation/Block.d

```diff
@@ -139,6 +149,58 @@ public string isInvalidReason (const ref Block block, Height prev_height,
             return fail_reason;
     }
 
+    if (getValidatorAtIndex !is null && active_enrollments > 0)
```

Comment:
This should be an assert. There must always be active enrollments?

Reply:
Also if `getValidatorAtIndex` is null then nothing is checked. This will have to be fixed.

Reply:
I removed this check so that the signatures are always checked.

Reply:
The check is still here?